### PR TITLE
Add initial message formats

### PR DESCRIPTION
**Pull request by tylerauerbeck**
Adds initial message formats and creates plain load balancer based upon structured event

Signed-off-by: Tyler Auerbeck <tylerauerbeck@users.noreply.github.com>

### CODE REVIEW

**Review by fishnix — COMMENTED** [review]
I'm thinking the idea here is that `pubsubx.Message` will be a generic message format for infratographer stuff and end up in a shared lib and `AdditionalData` will be the payload for the actual thing?  I just had a couple comments 👍

File: internal/srv/handlers.go [hunk 1]
```diff
@@ -1,56 +1,55 @@
...
-	switch m.Subject {
-	case fmt.Sprintf("%s.create", s.Prefix):
-		err := s.createMessageHandler(m)
-		if err != nil {
+	msg := pubsubx.Message{}
+	if err := json.Unmarshal(m.Data, &msg); err != nil {
+		s.Logger.Errorln("Unable to process data in message: %s", err)
+	}
+	switch msg.EventType {
+	case "create":
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply internal/srv/handlers.go]: ✅

File: pkg/pubsubx/message.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Package pubsubx provides common utilities and formats for working with pubsub systems
+package pubsubx
+
+import (
+	"time"
+)
+
+// Message contains the data structure expected to be received when picking
+// an event from a message queue
+type Message struct {
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply pkg/pubsubx/message.go:10]: You're right, this is going to end up as `pubsubx` in `https://github.com/infratographer/x`. Currently have a PR open over there if you don't mind taking this comment there? That way we make the fixes / get the feedback where it's going to live in the long run.

File: internal/srv/handlers.go [hunk 3]
```diff
@@ -1,56 +1,61 @@
...
 
-	"github.com/Moby/Moby/pkg/namesgenerator"
 	"github.com/nats-io/nats.go"
+
+	events "go.infratographer.sh/loadbalanceroperator/pkg/events/v1alpha1"
+	"go.infratographer.sh/loadbalanceroperator/pkg/pubsubx"
+)
+
+const (
+	EventCreate = "create"
```
Comment (sfunkhouser) [inline internal/srv/handlers.go:14]: should this come from the `events` package? Will we need it elsewhere?

File: internal/srv/app.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ func (s *Server) CreateNamespace(groupID string) error {
 // CreateApp deploys a loadBalancer based upon the configuration provided
 // from the event that is processed.
 func (s *Server) CreateApp(name string, chartPath string, namespace string) error {
-	releaseName := fmt.Sprintf("%s-%s", name, namespace)
+	releaseName := fmt.Sprintf("lb-%s-%s", name, namespace)
```
Comment (andy-v-h) [inline internal/srv/app.go:54]: This should be length checked and truncated, if greater than 63, ya?

**Review by sfunkhouser — APPROVED** [review]

File: internal/srv/handlers.go [hunk 5]
```diff
@@ -1,56 +1,61 @@
...
+	EventUpdate = "update"
```
[the remark this answers is not in the document]
Comment (tylerauerbeck) [reply internal/srv/handlers.go]: ✅

File: internal/srv/app.go (hunk 4 above)
Comment (tylerauerbeck) [reply internal/srv/app.go:58]: ✅

File: internal/srv/handlers.go (hunk 3 above)
Comment (tylerauerbeck) [reply internal/srv/handlers.go]: Yepp, I think this makes sense for now. Hoping to clean this up a bit more when we refactor some of this

**Review by andy-v-h — APPROVED** [review]

**Pull request merged**